utils/atomic-data-publisher/src/main.rs

```diff
@@ -0,0 +1,145 @@
+mod config;
+mod database;
+mod errors;
+mod metrics;
+mod protobuf;
+mod publisher;
+mod queries;
+mod service;
+
+use anyhow::{Context, Result};
+use config::{LoggingConfig, Settings};
+use service::AtomicDataPublisher;
+use std::sync::Arc;
+use tracing::{error, info, warn};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+  run_service().await
+}
+
+async fn run_service() -> Result<()> {
+  let settings = Settings::new().context("Failed to load configuration")?;
+
+  initialize_logging(&settings.logging)?;
+
+  info!("Starting Atomic Data Publisher");
+  info!("Configuration loaded successfully");
+
+  validate_config(&settings).context("Configuration validation failed")?;
+
+  let service = Arc::new(
```

Comment:
Does this need to be an Arc? It's just being run here and not referenced anywhere else